### PR TITLE
Build Go 1.21.4 and 1.20.11 images

### PR DESCRIPTION
**Pull request by cpanato**
#### What type of PR is this?

/kind feature

#### What this PR does / why we need it:

- Build Go 1.21.4 and 1.20.11 images for actively maintained Kubernetes release branches

#### Which issue(s) this PR fixes:

xref #3347

#### Does this PR introduce a user-facing change?
```release-note
Build Go 1.21.4 and 1.20.11 images
```

/assign @xmudrii  @saschagrunert  @jeremyrickard @puerco  
cc @kubernetes/release-engineering

### CODE REVIEW

File: dependencies.yaml [hunk 1]
```diff
@@ -231,7 +231,7 @@ dependencies:
       match: REVISION:\ '\d+'
 
   - name: "registry.k8s.io/build-image/kube-cross (v1.25-go1.20)"
-    version: v1.25.0-go1.20.10-bullseye.0
+    version: v1.25.0-go1.20.11-bullseye.0
```
Comment (xmudrii) [inline dependencies.yaml:234]: Should we change this given 1.25 has reached EOL?

Comment (cpanato) [reply dependencies.yaml:234]: oh yes, i thought is this month, but actually is eol :) will update

Comment (cpanato) [reply dependencies.yaml]: done

**Review by xmudrii — APPROVED** [review]
/lgtm
/approve

**Pull request merged**